addons/im_livechat/models/discuss_channel.py

```diff
@@ -237,3 +237,12 @@ def _chatbot_restart(self, chatbot_script):
             chatbot_script,
             Markup('<div class="o_mail_notification">%s</div>') % _('Restarting conversation...'),
         )
+
+    def _get_from_context_or_raise(self, thread_id):
+        try:
+            return super()._get_from_context_or_raise(thread_id)
+        except NotFound:
+            channel = self.search([("id", "=", thread_id), ("channel_type", "=", "livechat")], limit=1)
+            if channel and channel.check_access_rights("read", raise_exception=False):
```

Comment:
node: if using search (without sudo), it already includes the access rules and access rights, there is no need to call the method again.
also note that check_access_rights should always be called with check_access_rules to validate the full security set